### PR TITLE
Fix some misspellings in code, comments and xml docs.

### PR DESCRIPTION
**Pull request by dampir**
I've found and fixed some misspellings inside Octokit solution. There are a lot of changes, but they all very simple, one or two letter have been changed only. Hope this PR would be useful.

/cc @ryangribble, @shiftkey

### CODE REVIEW

File: Octokit.Tests.Integration/Clients/RepositoryForksClientTests.cs [hunk 1]
```diff
@@ -41,7 +40,7 @@ public class TheCreateMethod
             [IntegrationTest]
             public async Task ForkCreatedForUserLoggedIn()
             {
-                // The fork is created asynchronially by github and therefore it cannot 
+                // The fork is created asynchronically by github and therefore it cannot
```
Comment (shiftkey) [inline Octokit.Tests.Integration/Clients/RepositoryForksClientTests.cs]: `asynchronously`

File: Octokit.Tests.Integration/Clients/RepositoryForksClientTests.cs [hunk 2]
```diff
@@ -58,7 +57,7 @@ public class TheCreateMethod
             [OrganizationTest]
             public async Task ForkCreatedForOrganization()
             {
-                // The fork is created asynchronially by github and therefore it cannot 
+                // The fork is created asynchronically by github and therefore it cannot
```
Comment (shiftkey) [inline Octokit.Tests.Integration/Clients/RepositoryForksClientTests.cs]: `asynchronously`

File: Octokit/Models/Request/SearchRepositoriesRequest.cs [hunk 3]
```diff
@@ -479,9 +479,9 @@ public enum Language
         [SuppressMessage("Microsoft.Naming", "CA1704:IdentifiersShouldBeSpelledCorrectly", MessageId = "Edn")]
         Edn,
         Eiffel,
-        [SuppressMessage("Microsoft.Naming", "CA1704:IdentifiersShouldBeSpelledCorrectly", MessageId = "Elixer")]
-        [SuppressMessage("Microsoft.Naming", "CA1704:IdentifiersShouldBeSpelledCorrectly", MessageId = "Elixer")]
-        Elixer,
+        [SuppressMessage("Microsoft.Naming", "CA1704:IdentifiersShouldBeSpelledCorrectly", MessageId = "Elixir")]
+        [SuppressMessage("Microsoft.Naming", "CA1704:IdentifiersShouldBeSpelledCorrectly", MessageId = "Elixir")]
+        Elixir,
```
Comment (shiftkey) [inline Octokit/Models/Request/SearchRepositoriesRequest.cs]: These are going to be breaking changes for the unfortunate few which might be trying to use these - we need to leave the old ones around and mark them as obsolete

Comment (dampir) [inline Octokit/Models/Request/SearchRepositoriesRequest.cs]: I have one remarks about it. Yes, it is breaking changes, but this fully incorrect items (**Elixer**, Unified **Paralel ** C), these items never works correctly and if somebody use these then he doesn't  get any search results. Anyway, I reverted older items back and added new correct items. Hope they will be removed in further release.

**Comment by shiftkey** [comment]
Just a heads up: #1238 has a fix for the Travis issue - feel free to cherry-pick 8ec4bf388c674c1ce5f09f8c8ad88a470ce7f4d9 so you can verify this PR is still correct

**Comment by dampir** [comment]
@shiftkey, all checks, pls, check this out whenever you want :)

**Comment by shiftkey** [comment]
@dampir thanks!

**Pull request merged**